packages/platform-browser/src/hydration.ts

```diff
@@ -73,24 +68,28 @@ function hydrationFeature<FeatureKind extends HydrationFeatureKind>(
  * @publicApi
  * @developerPreview
  */
-export function withNoDomReuse(): HydrationFeature<HydrationFeatureKind.NoDomReuseFeature> {
+export function withNoDomReuse(): {ɵkind: HydrationFeatureKind.NoDomReuseFeature} {
   // This feature has no providers and acts as a flag that turns off
   // non-destructive hydration (which otherwise is turned on by default).
-  return hydrationFeature(HydrationFeatureKind.NoDomReuseFeature);
+  return {ɵkind: HydrationFeatureKind.NoDomReuseFeature};
 }
 
 /**
- * Disables HTTP transfer cache. Effectively causes HTTP requests to be performed twice: once on the
- * server and other one on the browser.
+ * @param options options for the transfer cache
+ * disable: Effectively causes HTTP requests to be performed twice: once on
+ * the server and other one on the browser.
+ * exclude: Callback function to determine if the request should be cached
  *
  * @publicApi
  * @developerPreview
  */
-export function withNoHttpTransferCache():
-    HydrationFeature<HydrationFeatureKind.NoHttpTransferCache> {
-  // This feature has no providers and acts as a flag that turns off
-  // HTTP transfer cache (which otherwise is turned on by default).
-  return hydrationFeature(HydrationFeatureKind.NoHttpTransferCache);
+export function withHttpTransferCacheOptions(
```

Comment:
I think we should keep the `withNoHttpTransferCache()` function and rename this function to `withHttpTransferCache({ ... } /* options */)`.